src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java

```diff
@@ -122,18 +131,76 @@ private boolean isVendorOwnershipCodeApplicableForTaxIdRoute(final VendorDetail
     private boolean isVendorTaxNumberInWorkday(final String vendorTaxNumber) {
         try {
             WorkdayKfsVendorLookupRoot result = getCuVendorWorkDayService().findEmployeeBySocialSecurityNumber(vendorTaxNumber, getDocumentNumber());
-            boolean isInWorkDay = result.isActiveEmployee();
-            LOG.debug("isVendorTaxNumberInWorkday, returning {} for document {}", isInWorkDay, getDocumentNumber());
-            return isInWorkDay;
+            final boolean isActiveOrTerminatedEmployee = isActiveOrTerminatedEmployeeWithinDateRange(result);
+            if (isActiveOrTerminatedEmployee) {
+                String message;
+                if (isActiveEmployee(result)) {
+                    message = getConfigurationService().getPropertyValueAsString(CUVendorKeyConstants.ACTIVE_EMPLOYEE_MESSAGE);
+                } else {
+                    message = MessageFormat.format(getConfigurationService().getPropertyValueAsString(CUVendorKeyConstants.TERMINATED_EMPLOYEE_MESSAGE),
+                            getEmployeeTerminationNumberOfDays());
+                }
+                annotateDocument(message);
+            }
+            LOG.debug("isVendorTaxNumberInWorkday, returning {} for document {}", isActiveOrTerminatedEmployee, getDocumentNumber());
+            return isActiveOrTerminatedEmployee;
         } catch (RuntimeException | URISyntaxException e) {
             LOG.error("isVendorTaxNumberInWorkday, got an error calling workday for document " + getDocumentNumber(), e);
-            annotateCouldNotCallWorkDay();
+            String message = getConfigurationService().getPropertyValueAsString(CUVendorKeyConstants.VENDOR_UNABLE_TO_CALL_WORKDAY);
+            annotateDocument(message);
+            return true;
+        }
+    }
+    
+    public boolean isActiveEmployee(WorkdayKfsVendorLookupRoot root) {
+        if (root != null) {
+            return root.getResults().stream().anyMatch(result -> result.isActive());
```

Comment:
For the is-active and is-active-or-terminated methods, it might be cleaner to simplify the conditions to something like this:

```java
return root != null && other_root_related_condition;
```

Reply:
ok